radix-engine-tests/tests/wasm_buffers.rs

```diff
@@ -0,0 +1,486 @@
+        _ => false,
+    });
+
+    // Act
+    let receipt = test_wasm_buffer_read!(
+        test_runner,
+        component_address,
+        read = (
+            256 * MB - 1, // SBOR max length
+            0,
+            get_memory_len(256 * MB - 1)
+        )
+    );
+
+    // Assert
+    receipt.expect_specific_failure(|e| match e {
+        RuntimeError::VmError(VmError::Wasm(WasmRuntimeError::ExecutionError(message))) => {
+            message == "Trap(Trap { reason: InstructionTrap(UnreachableCodeReached) })"
+        }
+        _ => false,
+    });
+}
+
+#[test]
+fn test_wasm_buffer_read_memory_size_too_large() {
+    // Arrange
+    let (mut test_runner, component_address) = get_test_runner();
+
+    // Act
+    let receipt = test_wasm_buffer_read!(
+        test_runner,
+        component_address,
+        read = (
+            256 * MB, // SBOR max length exceeded
+            0,
+            get_memory_len(256 * MB)
+        )
+    );
+
+    // Assert
+    receipt.expect_specific_failure(|e| match e {
+        RuntimeError::ApplicationError(ApplicationError::PanicMessage(message)) => {
+            message.contains("SizeTooLarge")
+        }
+        _ => false,
+    });
+
+    // Act
+    let receipt = test_wasm_buffer_read!(
+        test_runner,
+        component_address,
+        read = (usize::MAX as u64, 0, usize::MAX as u64)
+    );
+
+    // Assert
+    receipt.expect_specific_failure(|e| match e {
+        RuntimeError::ApplicationError(ApplicationError::PanicMessage(message)) => {
+            message.contains("SizeTooLarge")
+        }
+        _ => false,
+    });
+}
+
+#[test]
+fn test_wasm_buffer_invalid_buffer_id() {
+    // Arrange
+    let (mut test_runner, component_address) = get_test_runner();
+
+    for buffer_id in [0, 1, 3, u32::MAX] {
+        // Act
+        let receipt =
+            test_wasm_buffer_consume!(test_runner, component_address, buffer_id = buffer_id);
+        // Assert
+        receipt.expect_specific_failure(|e| {
+            matches!(
+                e,
+                RuntimeError::VmError(VmError::Wasm(WasmRuntimeError::BufferNotFound(..))),
+            )
+        });
+    }
+    // Act
+    let receipt = test_wasm_buffer_consume!(
+        test_runner,
+        component_address,
+        buffer_id = 2 // buffer_id=2 id of the kv_entry buffer
+    );
+    // Assert
+    receipt.expect_commit_success();
+}
+
+#[test]
+fn test_wasm_buffer_invalid_buffer_pointer() {
+    // Arrange
+    let (mut test_runner, component_address) = get_test_runner();
+    // Write 1KB to KV store
+    test_wasm_buffer_read!(
+        test_runner,
+        component_address,
+        read = (1 * KB, 0, get_memory_len(1 * KB))
+    );
+
+    // Act
+    let receipt = test_wasm_buffer_consume!(
+        test_runner,
+        component_address,
+        buffer_ptr = 0 // Invalid pointer
+    );
+    // Assert
+    receipt.expect_commit_success(); // This is actually success but should be failure
```

Comment:
We've got success when invalid pointer is provided. Not really sure if it should be commit succes.
From the WASM perspective the pointer is fine, it is within memory limits: `0 - memory.max`

Reply:
Agreed that this is acceptable for WASM.